cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -665,7 +665,7 @@ private ApplicationId updateApplicationByArtifact(ApplicationId appId,
 
     return updateApplicationInternal(appId, appSpec.getConfiguration(), programId -> { }, newArtifactDetail,
                                      Collections.singletonList(ApplicationConfigUpdateAction.UPGRADE_ARTIFACT),
-                                     allowedArtifactScopes, allowSnapshot, ownerAdmin.getOwner(appId));
+                                     allowedArtifactScopes, allowSnapshot, ownerAdmin.getOwner(appId), appSpec);
```

Comment:
Should we remove the second argument `appSpec.getConfiguration()` as it can be fetched from the appSpec itself? I don't think appSec can be null here as we are already calling methods on it without checking nullability